pkgs/by-name/do/doomsday-engine/package.nix

```
@@ -0,0 +1,101 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+
+  cmake,
+  ninja,
+  pkg-config,
+  python3,
+  makeBinaryWrapper,
+
+  SDL2,
+  SDL2_mixer,
+  the-foundation,
+  ncurses,
+  glbinding,
+  assimp,
+  glib,
+  qt5,
+  xorg,
+
+  nix-update-script,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "doomsday-engine";
+  version = "2.3.1";
+
+  src = fetchFromGitHub {
+    owner = "skyjake";
+    repo = "Doomsday-Engine";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-7jLL8ho3JL4amSTTu9ICmK1rcXIlJ9XWzh/rE6jT1zI=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+    python3
+    makeBinaryWrapper
+
+    qt5.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    SDL2
+    SDL2_mixer
+    the-foundation
+    ncurses
+    glbinding
+    assimp
+    glib
+
+    qt5.qtbase
+    qt5.qtx11extras
+    xorg.libXrandr
+    xorg.libXxf86vm
+  ];
+
+  # TODO: enable FMOD support by setting FMOD_DIR
+  cmakeFlags = [
+    (lib.cmakeBool "DENG_ASSIMP_EMBEDDED" false) # Use system assimp
+    (lib.cmakeFeature "DE_PREFIX" (placeholder "out")) # Legacy output prefix
+  ];
+
+  preConfigure = ''
+    # Doomsday builds PK3 files for these games by default - PK3 files are really just ZIP files,
+    # whose contents must have a modification date not earlier than 1980.
+    find doomsday/{apps/client/data,apps/plugins/{doom,heretic,hexen,doom64}/{defs,data}} \
+      -exec touch -d '1980-01-01T00:00:00Z' {} +
```

```suggestion
      -exec strip-nondeterminism --type zip {} +
```
This isn't in the stdenv iirc, so you'll need to pull in the `strip-nodeterminism` package